roachpb: fix RangeDescriptor generation counter printing

tbg opened the pull request:
The code was printing a `*int64` via `%d` which gives you the
address.

Release note: None

benesch commented on pkg/roachpb/metadata.go:
```diff
@@ -199,7 +199,11 @@ func (r RangeDescriptor) String() string {
+	gen := int64(-1)
+	if r.Generation != nil {
+		gen = *r.Generation
+	}
```
I think you just want to use `r.GetGeneration()` here. A nil `Generation` is considered 0 (not -1) and is how all new range descriptors start.

tbg committed:
roachpb: fix RangeDescriptor generation counter printing

The code was printing a `*int64` via `%d` which gives you the
address.

Release note: None

tschottdorf commented:
Done! TFTR
bors r=benesch
